extensions/functions_arithmetic.yaml

```diff
@@ -742,6 +742,107 @@ aggregate_functions:
           - value: fp64
         nullability: DECLARED_OUTPUT
         return: fp64?
+  - name: "median"
+    description: Calculate the median for a set of values.
+    impls:
+      - args:
+          - name: rounding
+            options: [ TIE_TO_EVEN, TIE_AWAY_FROM_ZERO, TRUNCATE, CEILING, FLOOR ]
+            required: false
+          - value: i8
+        nullability: DECLARED_OUTPUT
+        return: i8?
```

Comment:
I think this should be `fp64?`. There's no way to guarantee that the result is going to be an `i8`. Similarly for the remaining overloads as well as `approx_median`.

Reply:
There's no reason for the i8/i16/i32 variants to exist at all IMO. f(integer) -> integer variants for these statistical functions is not really wrong (you'd just have to round a lot more aggressively than you would for floats...) but make little sense in practice. i8/i16/i32 can be losslessly converted to fp32 and/or fp64 prior to applying this function whenever you actually meant to use floats. f(i8/i16/i32) -> fp64 is problematic because you have to pick one of the number types more or less at random (what if we ever add an fp128 type, do we now change the return type? see also #251). An argument could be made for f(i64) -> fp64 because it cannot be losslessly converted to an fp64 (and we did this for a similar function at some point, though I can't remember which off the top of my head), though it has that same problem in the end, and I doubt anyone's going to implement that in a way that doesn't just convert the i64 inputs to fp64 directly anyway.

Reply:
Scratch that. As @vibhatha pointed out offline, median is a bit special among the statistical functions because it returns one of the inputs exactly.

Reply:
A little doubt on the median, what if the data distribution is integer and why should we return floats in that case?